bin/garden-feat

```diff
@@ -0,0 +1,155 @@
+
+    # Switch to defined action
+    if args.type == 'cname':
+        # cname is a shortened name alias an only returns
+        # a single element of platform and flags
+        _platform = [k for k,v in features_matrix.items() if v['type'] in 'platform']
+        _flag = [k for k,v in features_matrix.items() if v['type'] in '_flag']
+        filtered_output = []
+        filtered_output.append(_platform[0])
+        filtered_output.append(_flag[0].lstrip('_'))
+        # This affects the artifacts filename
+        print('_'.join(str(s) for s in filtered_output))
+
+    elif args.type == 'features':
+        filter = ['platform', 'element', 'flag']
+        filtered_output = [k for k,v in features_matrix.items() if v['type'] in filter]
+        print(','.join(str(s) for s in filtered_output))
+
+    elif args.type == 'platforms':
+        filter = ['platform']
+        filtered_output = [k for k,v in features_matrix.items() if v['type'] in filter]
+        # Only one platform may be given
+        if len(filtered_output) > 1:
+            print(f'Error: Only a single platform is supported. Given: {filtered_output}')
+            sys.exit(1)
+        print(','.join(str(s) for s in filtered_output))
+
+    elif args.type == 'flags':
+        filter = ['flag']
+        filtered_output = [k for k,v in features_matrix.items() if v['type'] in filter]
+        print(','.join(str(s) for s in filtered_output))
+
+    elif args.type == 'elements':
+        filter = ['element']
+        filtered_output = [k for k,v in features_matrix.items() if v['type'] in filter]
+        print(','.join(str(s) for s in filtered_output))
+
+
+def get_args():
+    """ Get external cli args """
+    parser = argparse.ArgumentParser()
+
+    # Impositional arg(s)
+    parser.add_argument('--featureDir', type=str, default="../features")
+    parser.add_argument('--features', type=str, required=True)
+    parser.add_argument('--ignore', type=str, default=",")
+
+    # Positional arg(s)
+    args_type_allowed = [
+        'cname',
+        'features',
+        'platforms',
+        'flags',
+        'elements'
+    ]
+
+    parser.add_argument("type", nargs="?", choices=args_type_allowed, default="cname")
+    args = parser.parse_args()
+    return(args)
+
+
+def conv_features_list(features):
+    """ Convert comma seperated list to Python list """
+    if type(features) != 'list':
+        list_features = features.split(",")
+        return list_features
+
+
+def get_yaml_content(fname):
+    """ Read YAML files and get content """
+    try:
+        with open(fname) as f:
+            content = yaml.load(f, Loader=yaml.FullLoader)
+            return content
+    except FileNotFoundError:
+        # Avoid creating an unexpected artifact with missing
+        # features
+        print(f'Error: Could not find feature: {fname}')
+        sys.exit(1)
+
+
+def get_all_features(feature_dir, features, features_ignore, features_matrix):
```

Comment:
We should rewrite this function into two recursive functions to save iterations. We can have a small chat to discuss how this can be achieved.